pkg/api/pod/util.go

```diff
@@ -396,6 +432,7 @@ func GetValidationOptionsFromPodSpecAndMeta(podSpec, oldPodSpec *api.PodSpec, po
 		AllowInvalidLabelValueInSelector:                  false,
 		AllowInvalidTopologySpreadConstraintLabelSelector: false,
 		AllowMutableNodeSelectorAndNodeAffinity:           utilfeature.DefaultFeatureGate.Enabled(features.PodSchedulingReadiness),
+		AllowResizePolicy:                                 utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) || !hasNewOrUpdatedResizePolicy(podSpec, oldPodSpec),
```

Comment:
drop changes to this file... we don't need this in validation options, we clear the field before reaching validation if the feature is disabled